Testscripts/Linux/Linux-Kernel-Self-Tests.sh

```diff
@@ -0,0 +1,405 @@
+rtc sigaltstack size splice static_keys sync sysctl user x86 zram"
+
+# Skip these targets for they could hang or our scenario is not involved
+SKIP_TARGETS="android breakpoints cpu-hotplug memory-hotplug vm powerpc"
+
+# Skip these targets for they are failed to compile.
+SKIP_TARGETS_COMPILE_FAIL_CENTOS7="intel_pstate proc size x86"
+SKIP_TARGETS_COMPILE_FAIL_CENTOS8="size x86"
+SKIP_TARGETS_COMPILE_FAIL_UBUNTU16="drivers/dma-buf filesystems/binderfs proc"
+
+# Skip these targets for the test kernel are non-support
+SKIP_TARGETS_NONSUPPORT_UBUNTU16="membarrier"
+SKIP_TARGETS_NONSUPPORT_CENTOS7="drivers/dma-buf mount nsfs"
+SKIP_TARGETS_NONSUPPORT_CENTOS8="drivers/dma-buf"
+SKIP_TARGETS_NONSUPPORT_DEBIAN="drivers/dma-buf mount"
+
+# Ignorable these failed tests for the kernel are non-support or test cases are unstable
+IGNORABLE_FAIL_TESTS_UBUNTU16=("x86: test_syscall_vdso_32" "seccomp: seccomp_bpf")
+IGNORABLE_FAIL_TESTS_CENTOS_7=("seccomp: seccomp_bpf")
+IGNORABLE_FAIL_TESTS_CENTOS_8=("proc: setns-dcache" "proc: proc-pid-vm" "seccomp: seccomp_bpf")
+IGNORABLE_FAIL_TESTS_DEBIAN=("seccomp: seccomp_bpf")
+
+TOTAL_TARGETS=""
+CUSTOM_KERNEL_FLAG="FALSE"
+VERSION_ID=""
+
+ubuntu_src_git="git://git.launchpad.net/~canonical-kernel/ubuntu/+source/linux-azure/+git/"
+next_kernel_src="https://git.kernel.org/pub/scm/linux/kernel/git/next/linux-next.git"
+stable_kernel_src="git://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git"
+
+###############################################################################
+function install_dependencies() {
+    LogMsg "Installing dependency packages"
+
+    GetDistro
+    update_repos
+
+    case $DISTRO in
+        ubuntu* | debian*)
+            CheckInstallLockUbuntu
+            deb_packages=(make git gcc flex bison clang llvm fuse gcc-multilib libfuse2 \
+                        libc6-i386 libc6-dev-i386 libelf-dev libcap-ng-dev libfuse-dev \
+                        libpopt-dev libnuma-dev libmount-dev libcap-dev build-essential \
+                        pkg-config bc rsync)
+            LogMsg "Dependency package names： ${deb_packages[@]}"
+            install_package "${deb_packages[@]}" >> $BUILDING_LOG 2>&1
+            ;;
+        centos_7 | centos_8 | redhat_7 | redhat_8)
+            rpm_packages=(make git gcc flex bison clang llvm fuse libcap-ng-devel popt-devel \
+                        libcap-devel glibc-devel.*i686 fuse-devel elfutils-devel \
+                        numactl-devel glibc-devel)
+            if [[ $DISTRO != centos_8 ]]; then
+                rpm_packages+=(libmount-devel)
+            fi
+            install_epel
+            LogMsg "Dependency package names： ${rpm_packages[@]}"
+            install_package "${rpm_packages[@]}" >> $BUILDING_LOG 2>&1
+            ;;
+        *)
+            LogErr "Unsupported distro: $DISTRO"
+            SetTestStateSkipped
+            exit 0
+            ;;
+    esac
+}
+
+function download_custom_kernel() {
+    cd /root
```

Comment:
what VM size have you used? and distro? don't you have disk full error in RHEL 7.7?

Reply:
This test case's VM size is Standard_DS4_v2. I have /home/lisa full error in RHEL 7.7. So I changed into /root.